pkg/render/fluentd_test.go

```diff
@@ -917,104 +920,133 @@ var _ = Describe("Tigera Secure Fluentd rendering tests", func() {
+		Expect(envs).To(ContainElement(corev1.EnvVar{Name: "TENANT_ID", Value: "test-tenant-id"}))
+		Expect(envs).To(ContainElement(corev1.EnvVar{Name: "LINSEED_TOKEN", Value: "/var/run/secrets/kubernetes.io/serviceaccount/token"}))
+	})
+
+	It("should render with EKS Cloudwatch Log for managed cluster with linseed token volume", func() {
+
+		expectedResources := getExpectedResourcesForEKS()
+
+		// Add linseed role binding to read linseed secrets
+		linseedResources := []struct {
 			name    string
 			ns      string
 			group   string
 			version string
 			kind    string
 		}{
-			{name: "tigera-fluentd", ns: "", group: "", version: "v1", kind: "Namespace"},
-			{name: render.FluentdPolicyName, ns: render.LogCollectorNamespace, group: "projectcalico.org", version: "v3", kind: "NetworkPolicy"},
-			{name: render.FluentdMetricsService, ns: render.LogCollectorNamespace, group: "", version: "v1", kind: "Service"},
-			{name: "eks-log-forwarder", ns: "", group: "rbac.authorization.k8s.io", version: "v1", kind: "ClusterRole"},
-			{name: "eks-log-forwarder", ns: "", group: "rbac.authorization.k8s.io", version: "v1", kind: "ClusterRoleBinding"},
-			{name: "eks-log-forwarder", ns: "", group: "policy", version: "v1beta1", kind: "PodSecurityPolicy"},
-			{name: "eks-log-forwarder", ns: "tigera-fluentd", group: "", version: "v1", kind: "ServiceAccount"},
-			{name: "tigera-eks-log-forwarder-secret", ns: "tigera-fluentd", group: "", version: "v1", kind: "Secret"},
-			{name: "eks-log-forwarder", ns: "tigera-fluentd", group: "apps", version: "v1", kind: "Deployment"},
-			{name: "tigera-fluentd", ns: "", group: "rbac.authorization.k8s.io", version: "v1", kind: "ClusterRole"},
-			{name: "tigera-fluentd", ns: "", group: "rbac.authorization.k8s.io", version: "v1", kind: "ClusterRoleBinding"},
-			{name: "tigera-fluentd", ns: "", group: "policy", version: "v1beta1", kind: "PodSecurityPolicy"},
-			{name: "fluentd-node", ns: "tigera-fluentd", group: "", version: "v1", kind: "ServiceAccount"},
-			{name: render.PacketCaptureAPIRole, ns: render.LogCollectorNamespace, group: "rbac.authorization.k8s.io", version: "v1", kind: "Role"},
-			{name: render.PacketCaptureAPIRoleBinding, ns: render.LogCollectorNamespace, group: "rbac.authorization.k8s.io", version: "v1", kind: "RoleBinding"},
-			// Daemonset
-			{name: "fluentd-node", ns: "tigera-fluentd", group: "apps", version: "v1", kind: "DaemonSet"},
+			{name: "tigera-linseed", ns: render.LogCollectorNamespace, group: "rbac.authorization.k8s.io", version: "v1", kind: "RoleBinding"},
 		}
 
-		fetchInterval := int32(900)
-		cfg.EKSConfig = &render.EksCloudwatchLogConfig{
-			AwsId:         []byte("aws-id"),
-			AwsKey:        []byte("aws-key"),
-			AwsRegion:     "us-west-1",
-			GroupName:     "dummy-eks-cluster-cloudwatch-log-group",
-			FetchInterval: fetchInterval,
-		}
+		// we need to insert the linseedResource at index 11 to assert the resources in order
```

Comment:
This type of test is very very brittle, and is going to result in unnecessary failures when we modify the Linseed render code in the future for unrelated changes that happen to adjust resource ordering.

We should be using this helper function here instead: https://github.com/caseydavenport/operator/blob/3919c836a4518f6ff5569fcef9551f99930f3933/pkg/render/common/test/testing.go#L93-L105

It performs an order agnostic check and emits much better diagnostic output on failure.